### ISSUE TITLE
twisted.mail.smtp.SMTP incorrectly announces itself as ESMTP

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#1891]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#1891|
|Type|defect|
|Created|2006-07-02 21:14:23Z|

<details><summary>Searchable metadata</summary>

```
trac-id__1891 1891
type__defect defect
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__mail mail
keywords__ 
time__1151874863000000 1151874863000000
changetime__1151980401000000 1151980401000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

Ready for review in esmtp-announcement-1891

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml set owner to @jml<br>@jml set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml set owner to @exarkun<br>@jml set status to `new`|
|-|-|


You seem to be putting two lines of whitespace between functions, rather than one.  It's not a violation of any written standards, but it doesn't seem to be the done thing.

The docstrings on the tests are better than nothing.  They would be better if they explained why ESMTP should not appear in the greeting. The reason why could be nothing more than a link to the relevant RFC or ticket.

I'm happy for this to merge even if you don't make these changes, providing all the tests pass.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set status to `closed`|
|-|-|

(In [17470]) Merge esmtp-announcement-1891

Author: exarkun
Reviewer: jml
Fixes [#1891](https://github.com/twisted/twisted/issues/1891)

Add tests for the server greetings sent by SMTP and ESMTP and fix them
so that only ESMTP greets with an "ESMTP" string.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
